code/modules/antagonists/creep/creep.dm

```diff
@@ -155,55 +155,30 @@
 	else
 		explanation_text = "Free Objective"
 
+
 /datum/objective/assassinate/jealous/proc/find_coworker(datum/mind/oldmind)//returning null = free objective
-	if(!oldmind.assigned_role)
+	if(is_unassigned_job(oldmind.assigned_role))
 		return
 	var/list/viable_coworkers = list()
 	var/list/all_coworkers = list()
-	var/chosen_department
-	var/their_chosen_department
-	//note that command and sillycone are gone because borgs can't be obsessions and the heads have their respective department. Sorry cap, your place is more with centcom or something
-	if(oldmind.assigned_role in GLOB.security_positions)
-		chosen_department = "security"
-	if(oldmind.assigned_role in GLOB.engineering_positions)
-		chosen_department = "engineering"
-	if(oldmind.assigned_role in GLOB.medical_positions)
-		chosen_department = "medical"
-	if(oldmind.assigned_role in GLOB.science_positions)
-		chosen_department = "science"
-	if(oldmind.assigned_role in GLOB.supply_positions)
-		chosen_department = "supply"
-	if(oldmind.assigned_role in GLOB.service_positions)
-		chosen_department = "service"
-	for(var/mob/living/carbon/human/H in GLOB.alive_mob_list)
-		if(!H.mind)
+	var/our_departments = oldmind.assigned_role.departments
+	for(var/mob/living/carbon/human/human_alive in GLOB.alive_mob_list)
+		if(!human_alive.mind)
 			continue
-		if(!SSjob.GetJob(H.mind.assigned_role) || H == oldmind.current || H.mind.has_antag_datum(/datum/antagonist/obsessed))
+		if(human_alive == oldmind.current || human_alive.mind.assigned_role.faction != "Station" || human_alive.mind.has_antag_datum(/datum/antagonist/obsessed))
```

Comment:
```suggestion
		if(human_alive == oldmind.current || human_alive.mind.assigned_role.faction != FACTION_STATION || human_alive.mind.has_antag_datum(/datum/antagonist/obsessed))
```

Reply:
What are you pointing at in here?

Reply:
The `FACTION_STATION` macro can be used here instead of `"Station"` I believe?

Reply:
Either you're confused by the git diff notation, or I am, because that's precisely the change that is happening here, isn't it?

Reply:
![image](https://user-images.githubusercontent.com/51863163/124346932-34652f80-dba7-11eb-8b82-4a1d6daed26c.png)
This is what I see in the file, the change above is a code suggestion perhaps I made it unclear, and I do apologize for that!

Reply:
Oh, it was one of the two options, and I was the one confused.
I'll implement this, thanks for the review!